### PR TITLE
migrate /cluster logs into structured logging

### PR DESCRIPTION
**Pull request by CKchen0726**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
part of kubernetes/enhancements#1602

Which issue(s) this PR fixes:
keps/sig-instrumentation/1602-structured-logging
Structured Logging migration instructions

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
```release-note
migrate /cluster logs into structured logging
```

### CODE REVIEW

File: cluster/images/etcd/migrate/data_dir.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func OpenOrCreateDataDirectory(path string) (*DataDirectory, error) {
 		return nil, err
 	}
 	if !exists {
-		klog.Infof("data directory '%s' does not exist, creating it", path)
+		klog.InfoS("data directory does not exist, creating it", "path", path)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/data_dir.go:47]:
```suggestion
		klog.InfoS("Data directory does not exist, creating it", "path", path)
```

File: cluster/images/etcd/migrate/data_dir.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func (d *DataDirectory) Initialize(target *EtcdVersionPair) error {
 		return err
 	}
 	if isEmpty {
-		klog.Infof("data directory '%s' is empty, writing target version '%s' to version.txt", d.path, target)
+		klog.InfoS("data directory is empty, writing target version to version.txt", "path", d.path, "targetVersion", target)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/data_dir.go:69]:
```suggestion
		klog.InfoS("Data directory is empty, writing target version to version.txt", "path", d.path, "targetVersion", target)
```

File: cluster/images/etcd/migrate/migrate_server.go [hunk 3]
```diff
@@ -115,19 +115,19 @@ func (r *EtcdMigrateServer) Stop() error {
 		case <-stopped:
 			return
 		case <-timedout:
-			klog.Infof("etcd server has not terminated gracefully after %s, killing it.", gracefulWait)
+			klog.InfoS("etcd server has not terminated gracefully after a while, killing it.", "timeout", gracefulWait)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/migrate_server.go:118]:
```suggestion
			klog.InfoS("Etcd server has not terminated gracefully after a while, killing it.", "timeout", gracefulWait)
```

File: cluster/images/etcd/migrate/migrate_server.go [hunk 4]
```diff
@@ -115,19 +115,19 @@ func (r *EtcdMigrateServer) Stop() error {
...
 			r.cmd.Process.Kill()
 			return
 		}
 	}()
 	err = r.cmd.Wait()
 	stopped <- true
 	if exiterr, ok := err.(*exec.ExitError); ok {
-		klog.Infof("etcd server stopped (signal: %s)", exiterr.Error())
+		klog.InfoS("etcd server stopped with signal", "signal", exiterr.Error())
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/migrate_server.go:126]:
```suggestion
		klog.InfoS("Etcd server stopped with signal", "signal", exiterr.Error())
```

File: cluster/images/etcd/migrate/migrator.go [hunk 5]
```diff
@@ -84,11 +84,11 @@ func (m *Migrator) MigrateIfNeeded(target *EtcdVersionPair) error {
 	}
 
 	for {
-		klog.Infof("Converging current version '%s' to target version '%s'", current, target)
+		klog.InfoS("Converging current version to target version", "currentVersion", current, "targetVersion", target)
 		currentNextMinorVersion := &EtcdVersion{Version: semver.Version{Major: current.version.Major, Minor: current.version.Minor + 1}}
 		switch {
 		case current.version.MajorMinorEquals(target.version) || currentNextMinorVersion.MajorMinorEquals(target.version):
-			klog.Infof("current version '%s' equals or is one minor version previous of target version '%s' - migration complete", current, target)
+			klog.InfoS("current version equals or is one minor version previous of target version - migration complete", "currentVersion", current, "targetVersion", target)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/migrator.go:91]:
```suggestion
			klog.InfoS("Current version equals or is one minor version previous of target version - migration complete", "currentVersion", current, "targetVersion", target)
```

File: cluster/images/etcd/migrate/migrator.go [hunk 6]
```diff
@@ -100,10 +100,10 @@ func (m *Migrator) MigrateIfNeeded(target *EtcdVersionPair) error {
 			return fmt.Errorf("downgrading from etcd 3.x to 2.x is not supported")
 		case current.version.Major == target.version.Major && current.version.Minor < target.version.Minor:
 			stepVersion := m.cfg.supportedVersions.NextVersionPair(current)
-			klog.Infof("upgrading etcd from %s to %s", current, stepVersion)
+			klog.InfoS("upgrading etcd from current version to next version", "currentVersion", current, "nextVersion", stepVersion)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/migrator.go:103]:
```suggestion
			klog.InfoS("Upgrading etcd from current version to next version", "currentVersion", current, "nextVersion", stepVersion)
```

File: cluster/images/etcd/migrate/migrator.go [hunk 7]
```diff
@@ -100,10 +100,10 @@ func (m *Migrator) MigrateIfNeeded(target *EtcdVersionPair) error {
...
 			current, err = m.minorVersionUpgrade(current, stepVersion)
 		case current.version.Major == 3 && target.version.Major == 3 && current.version.Minor > target.version.Minor:
-			klog.Infof("rolling etcd back from %s to %s", current, target)
+			klog.InfoS("rolling etcd back from current version to target version", "currentVersion", current, "targetVersion", target)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/migrator.go:106]:
```suggestion
			klog.InfoS("Rolling etcd back from current version to target version", "currentVersion", current, "targetVersion", target)
```

File: cluster/images/etcd/migrate/options.go [hunk 8]
```diff
@@ -116,7 +116,7 @@ func lookupEnv(env string) (string, error) {
 }
 
 func fallbackToEnv(flag, env string) (string, error) {
-	klog.Infof("--%s unset - falling back to %s variable", flag, env)
+	klog.InfoS("command flag with prefix '--' unset - falling back to environment variable", "flag", flag, "environmentVariable", env)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/options.go:119]:
```suggestion
	klog.InfoS("Command flag with prefix '--' unset - falling back to environment variable", "flag", flag, "environmentVariable", env)
```

File: cluster/images/etcd/migrate/migrator.go [hunk 9]
```diff
@@ -131,14 +131,14 @@ func (m *Migrator) rollbackEtcd3MinorVersion(current *EtcdVersionPair, target *E
 
 	// Start current version of etcd.
 	runner := m.newServer()
-	klog.Infof("Starting etcd version %s to capture rollback snapshot.", current.version)
+	klog.InfoS("Starting etcd version to capture rollback snapshot.", "currentVersion", current.version)
 	err = runner.Start(current.version)
 	if err != nil {
 		klog.Fatalf("Unable to automatically downgrade etcd: starting etcd version %s to capture rollback snapshot failed: %v", current.version, err)
```
Comment (yangjunmyfm192085) [inline cluster/images/etcd/migrate/migrator.go:137]: Can you complete the modification of the whole file?

**Comment by yangjunmyfm192085** [comment]
Hi, @CKchen0726, Can you modify it to the correct format?

File: cluster/images/etcd/migrate/migrator.go (hunk 9 above)
Comment (CKchen0726) [reply cluster/images/etcd/migrate/migrator.go:137]: Ok, I will fix later.

**Comment by justaugustus** [comment]
bump for @kubernetes/wg-structured-logging-reviews

**Comment by dims** [comment]
@CKchen0726 looks like there are still pending suggestions.

**Comment by dims** [comment]
101132
107135
104490
103535
104192
104255
98729
104818
104800
104791
102742
100851
99308
105919
97021
88172
78837
103188
102769
106712
104753
105813
105590
99891
96936
105903
103486
87399
105760
100600
102844
105257
106623
106078
103701
97214
96275
106479
103128
104835
104374
103024
104356
103916
103740
103322
106778
101750
105282
102307
104398
98164

**Comment by dims** [comment]
This PR is older than 4 weeks and needs as rebase.

Please rebase the PR against latest master and reopen if still needed.

/close

**Pull request closed without merging**